src/lib/generate-key.cpp

```diff
@@ -91,62 +84,40 @@ load_generated_g10_key(pgp_key_t *           dst,
     // G10 always needs pubkey here
     assert(pubkey);
 
-    if (init_mem_dest(&memdst, NULL, 0)) {
-        goto end;
+    // this would be better on the stack but the key store does not allow it
+    std::unique_ptr<rnp_key_store_t> key_store(new (std::nothrow) rnp_key_store_t(ctx));
+    if (!key_store) {
+        return false;
     }
-
-    if (!g10_write_seckey(&memdst, newkey, NULL, ctx.rng)) {
+    /* Write g10 seckey */
+    rnp::MemoryDest memdst(NULL, 0);
+    if (!g10_write_seckey(&memdst.dst(), newkey, NULL, ctx.rng)) {
         RNP_LOG("failed to write generated seckey");
-        goto end;
-    }
-
-    // this would be better on the stack but the key store does not allow it
-    try {
```

Comment:
@andrey-utkin All exceptions are now catched in the functions which calls this one, and actually the only thing it could throw is allocation failure, unless I'm missing something. So less code lines, better coverage, easier code reading.